### PR TITLE
Tsk 61

### PR DESCRIPTION
**Pull request by MLengl**
imtoduced classificationinterface
improved unittests

### CODE REVIEW

File: lib/taskana-cdi/src/test/java/pro/taskana/TaskanaRestTest.java [hunk 1]
```diff
@@ -23,12 +29,12 @@
...
 
 	@GET
-	public Response startTask() throws NotAuthorizedException, WorkbasketNotFoundException, ClassificationNotFoundException {
+	public Response startTask() throws NotAuthorizedException, WorkbasketNotFoundException, ClassificationNotFoundException, ClassificationAlreadyExistException {
 		Workbasket workbasket = new Workbasket();
 		workbasket.setName("wb");
 		taskanaEjb.getWorkbasketService().createWorkbasket(workbasket);
-		Classification classification = new Classification();
-		taskanaEjb.getClassificationService().addClassification(classification);
+		Classification classification = (Classification) new ClassificationImpl();
```
Comment (BerndBreier) [inline lib/taskana-cdi/src/test/java/pro/taskana/TaskanaRestTest.java]: We should behave like a client in this situation. I.e. don't use ClassificationImpl but create a classification via ClassificationService.newClassification().

File: lib/taskana-core/src/main/java/pro/taskana/exceptions/ClassificationNotFoundException.java [hunk 2]
```diff
@@ -1,6 +1,7 @@
 package pro.taskana.exceptions;
 
 /**
+<<<<<<< HEAD
  * This exception will be thrown if a specific task is not in the database.
  */
```
Comment (BerndBreier) [inline lib/taskana-core/src/main/java/pro/taskana/exceptions/ClassificationNotFoundException.java]: Javadoc should say: ..thrown if a specific Classification is not in the database

File: lib/taskana-core/src/main/java/pro/taskana/impl/ClassificationServiceImpl.java [hunk 3]
```diff
@@ -72,17 +75,30 @@ public ClassificationServiceImpl(TaskanaEngine taskanaEngine, ClassificationMapp
...
+        try {
             taskanaEngineImpl.openConnection();
-            classification.setId(IdGenerator.generateWithPrefix(ID_PREFIX_CLASSIFICATION));
-            classification.setCreated(Date.valueOf(LocalDate.now()));
-
-            this.setDefaultValues(classification);
-
-            classificationMapper.insert(classification);
-            LOGGER.debug("Method addClassification added classification {}.", classification);
+            ClassificationImpl classificationImpl = (ClassificationImpl) classification;
```
Comment (BerndBreier) [inline lib/taskana-core/src/main/java/pro/taskana/impl/ClassificationServiceImpl.java]: we obtain a Classification from the client. Before we continue, we should ensure that it is a valid object. This implies for instance that the Id is set ...

Comment (MLengl) [reply lib/taskana-core/src/main/java/pro/taskana/impl/ClassificationServiceImpl.java]: setDefaultValues() is called before the insert-call. setting an ID if is/was null is included there.

File: lib/taskana-cdi/src/test/java/pro/taskana/TaskanaRestTest.java (hunk 1 above)
Comment (MLengl) [reply lib/taskana-cdi/src/test/java/pro/taskana/TaskanaRestTest.java]: I will search and fix all classification-casts which are like this. The instantiation by service-call will be added in the upcoming pull-request.

**Pull request merged**